src/main/kotlin/it/pagopa/wallet/scheduler/repositories/WalletRepository.kt

```diff
@@ -0,0 +1,23 @@
+package it.pagopa.wallet.scheduler.repositories
+
+import it.pagopa.wallet.scheduler.documents.Wallet
+import org.springframework.data.mongodb.repository.Aggregation
+import org.springframework.data.repository.reactive.ReactiveCrudRepository
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+
+@Repository
+interface WalletRepository : ReactiveCrudRepository<Wallet, String> {
+
+    @Aggregation(
+        "{\$match: {'creationDate': {'\$gte': '?0','\$lte': '?1'},'status': '?2'}}",
+        "{\$sort: {'updateDate': 1}}",
```

Comment:
Sorting should be done on creationDate to ensure consistency of the output

Reply:
Currently the index on creationDate is missing. Mongo does not allow the query without it

Reply:
What about walletId?

Reply:
These are the actual indexes of payment-wallets:

- _id
- updateDate
- contractId
- userId

Reply:
Order changed to creationDate here 0e66d26496154c650d7e5a168b073877245a719a